### DISCUSSION TITLE
How to restart i3 forgetting previous layout without forcing a crash

### DISCUSSION THREAD

**Discussion by kuben-joz** [i3/i3#5814] [Q&A]
Hi,
I'm trying to get the following layout change to work:
- Two monitors, built in laptop display and an external display
- I disable the laptop display

I have a problem whereby it remembers which workspaces where on the laptop display and keeps them there. Restarting in place doesn't work.

The only workaround I found was to switch to one of the workspaces on the disabled display and try to move it to the other monitor, this causes i3 to crash and gives me an option to 'restart i3 forgetting previous layout'. Can this restart be triggered without crashing i3?

**Comment by orestisfl** [comment]
Running something like `xrandr --auto` should re-configure your displays.

That said, i3 shouldn't crash. Can you upload a crash dump and your i3 version? https://i3wm.org/docs/debugging.html#_on_crashes_obtaining_a_backtrace

**Comment by sfischme** [comment]
I have the same problem. Here's the [initial post](https://github.com/i3/i3/discussions/5581) and the submitted [bug report](https://github.com/i3/i3/issues/5612).  Unfortunately I haven't found a fix yet.
